File: shell-java/src/main/java/com/dozeboy/sample1/MainActivity.java
package com.dozeboy.sample1;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;

import com.blankj.utilcode.util.LogUtils;
import com.dozeboy.sample1.base.BaseSampleActivity;
import com.dozeboy.sample1.databinding.ActivityMainBinding;

/**
 * @author dozeboy
 * @date 2019/1/6
 */
public class MainActivity extends BaseSampleActivity<ActivityMainBinding> {
    private static final String TAG = "MainActivity";
    private static Context context;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding.textHello.setOnClickListener(v -> {
            startActivity(new Intent(this, MainActivity.class));
        });
        LogUtils.d(this);
        context = this;
    }

    @Override
    protected void onDestroy() {
        LogUtils.d("onDestroy");
        super.onDestroy();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }
}

File: shell-kt/src/main/java/com/dozeboy/android/template/MainActivity.kt
package com.dozeboy.android.template

import android.os.Bundle
import androidx.appcompat.app.ActionBarDrawerToggle
import com.dozeboy.android.template.base.BaseActivity
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.toolbar.*

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupDrawer()
    }

    override fun getLayoutId(): Int = R.layout.activity_main
    private fun setupDrawer() {
        val toggle = ActionBarDrawerToggle(this, drawer_layout, toolbar, R.string.drawer_open, R.string.drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()
    }
}
